lib/active_model/serializer.rb

```diff
@@ -151,9 +162,9 @@ def success?
     #     serializer.as_json(include: { posts: { include: { comments: { only: :body } }, only: :title } })
     def serializable_hash(adapter_opts = nil)
       adapter_opts ||= {}
-      adapter_opts = { include: '*', adapter: :attributes }.merge!(adapter_opts)
-      adapter = ActiveModelSerializers::Adapter.create(self, adapter_opts)
-      adapter.serializable_hash(adapter_opts)
+      adapter_opts = { include: '*' }.merge!(adapter_opts)
+      adapter_instance = ActiveModelSerializers::Adapter::Attributes.new(self, adapter_opts)
+      serialize(adapter_opts, {}, adapter_instance)
```

Comment:
It really irritates me that this requires an adapter_instance (else I get a stack overflow).  

The adapter_instance interface is

- cached_name : used in making cache key
- class : used by fragment caching in fragment serializers
- fragment_cache : used by fragment caching